example/gin/server.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/go-swagno/swagno"
+	"github.com/go-swagno/swagno-gin/swagger"
+	"github.com/go-swagno/swagno/components/endpoint"
+	"github.com/go-swagno/swagno/components/mime"
+	"github.com/go-swagno/swagno/components/parameter"
+	"github.com/go-swagno/swagno/example/models"
+	"github.com/go-swagno/swagno/http/response"
+)
+
+func main() {
+	sw := swagno.New(swagno.Config{Title: "Testing API", Version: "v1.0.0"})
+	desc := "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed id malesuada lorem, et fermentum sapien. Vivamus non pharetra risus, in efficitur leo. Suspendisse sed metus sit amet mi laoreet imperdiet. Donec aliquam eros eu blandit feugiat. Quisque scelerisque justo ac vehicula bibendum. Fusce suscipit arcu nisl, eu maximus odio consequat quis. Curabitur fermentum eleifend tellus, lobortis hendrerit velit varius vitae."
+
+	endpoints := []*endpoint.EndPoint{
+		endpoint.New(
+			endpoint.WithMethod(endpoint.GET),
+			endpoint.WithPath("/product"),
+			endpoint.WithTags("product"),
+			endpoint.WithSuccessfulReturns([]response.Info{response.New([]models.Product{}, "200", "Product List")}),
+			endpoint.WithDescription(desc),
+			endpoint.WithProduce([]mime.MIME{mime.JSON, mime.XML}),
+			endpoint.WithConsume([]mime.MIME{mime.JSON}),
+			endpoint.WithSummary("this is a test summary"),
+		),
+		endpoint.New(
+			endpoint.WithMethod(endpoint.GET),
+			endpoint.WithPath("/product/{id}"),
+			endpoint.WithTags("product"),
+			endpoint.WithParams(parameter.IntParam("id", parameter.WithIn(parameter.Path), parameter.WithRequired())),
+			endpoint.WithSuccessfulReturns([]response.Info{models.SuccessfulResponse{}}),
+			endpoint.WithErrors([]response.Info{models.UnsuccessfulResponse{}}),
+			endpoint.WithProduce([]mime.MIME{mime.JSON, mime.XML}),
+		),
+		endpoint.New(
+			endpoint.WithMethod(endpoint.GET),
+			endpoint.WithPath("/product/{id}/detail"),
+			endpoint.WithTags("product"),
+			endpoint.WithParams(parameter.IntParam("id", parameter.WithIn(parameter.Path), parameter.WithRequired())),
+			endpoint.WithSuccessfulReturns([]response.Info{response.New(models.MapTest{
+				"data": models.Product{},
+			}, "200", "")}),
+			endpoint.WithErrors([]response.Info{response.New(map[string]interface{}{
```

Comment:
I like the flexibility to do it this way as well, but we'll need to update this so that when it renders on the UI it doesn't show `"error":"string"` and `"errorCode":0`. That's not really helpful, and would be much more helpful if it showed the values that you have in the struct.

Reply:
Also, if we don't have time to do this now as it will probably require an update to the generation logic we can make a todo and comeback to it later.

Reply:
Alright, I will work on this